EIPS/eip-4824.md

```diff
@@ -0,0 +1,220 @@
+---
+eip: 4824
+title: Decentralized Autonomous Organizations
+description: A standard URI and JSON schema for decentralized autonomous organizations (DAOs).
+author: Joshua Tan (@thelastjosh), Isaac Patka (@ipatka), Ido Gershtein <ido@daostack.io>, Eyal Eithcowich <eyal@deepdao.io>, Michael Zargham (@mzargham), Sam Furter (@nivida)
+discussions-to: https://ethereum-magicians.org/t/eip-4824-decentralized-autonomous-organizations/8362
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-02-17
+---
+
+## Abstract
+
+A standard URI and JSON schema for decentralized autonomous organizations (DAOs), focusing on relating on-chain and off-chain representations of membership and proposals.
+
+## Motivation
+
+DAOs, since being invoked in the Ethereum whitepaper, have been vaguely defined. This has led to a wide range of patterns but little standardization or interoperability between the frameworks and tools that have emerged. Standardization and interoperability are necessary to support a variety of use-cases. In particular, a standard daoURI, similar to tokenURI in [ERC-721](/EIPS/eip-721), will enhance DAO search, discoverability, legibility, and proposal simulation. More consistent data across the ecosystem is also a prerequisite for future DAO standards.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Every EIP-4824 compliant contract MUST implement the EIP4824 interface below:
+
+```solidity
+pragma solidity ^0.4.20;
+
+/// @title EIP-4824 DAOs
+/// @dev See https://eips.ethereum.org/EIPS/eip-4824
+
+interface EIP4824 {
+    /// @notice A distinct Uniform Resource Identifier (URI) pointing to a JSON object following the "EIP-4824 DAO JSON-LD Schema". This JSON file splits into four URIs: membersURI, proposalsURI, activityLogURI, and governanceURI. The membersURI should point to a JSON file that conforms to the "EIP-4824 Members JSON-LD Schema". The proposalsURI should point to a JSON file that conforms to the "EIP-4824 Proposals JSON-LD Schema". The activityLogURI should point to a JSON file that conforms to the "EIP-4824 Activity Log JSON-LD Schema". The governanceURI should point to a flatfile, normatively a .md file. Each of the JSON files named above can be statically-hosted or dynamically-generated.
+    function daoURI() external view returns (string _daoURI);
```

Comment:
Does it mean the caller of contract will just call `EIP4824 .daoURI` and then get a JSON schema that has other? 

Can you elaborate if considered making all the metadata available in the form of a Solidity `interface`? eg ERC-721

Reply:
Or if this is intentional that daoURI will return offchain information in JSON-schema, maybe update the name of this proposal to `Off-Chain metadata standard for DAOs`

Reply:
> Can you elaborate if considered making all the metadata available in the form of a Solidity interface? eg ERC-721
Interesting thought. To be clear, you're suggesting we make the `name` and `description` fields (but probably not `type`) available directly in the Solidity interface. We talked a bit about standardizing certain data on-chain, mainly around proposals, but never discussed standardizing these metadata fields directly on-chain.

My initial thought here is that having this basic "storefront" data in the JSON object covers the legibility, discoverability, and reporting use-cases that we want to support, and that it is marginally more elegant and easier for the DAO end-user just to have to set daoURI in the contract interface itself. I can kind of imagine use-cases where a contract just wants the name and description without having to call daoURI / interact with the outside world, but generally speaking the off-chain data will be fresher. Side note: ERC-721 puts just `name` and `symbol` in the Solidity interface while putting `name` and `description` in the off-chain JSON schema, so it's also not totally consistent or clear what fields should end up on-chain vs off-chain. Do you have any idea why they made those decisions?

Reply:
Re: proposed name change. We'd prefer to keep the current title.

As the EIP states, DAOs have been vaguely defined. 10 experts walk into a room with 10 definitions of what a DAO is. One of the things that came out of the cross-industry conversations (see the 'Community Consensus' section) that generated the standard was the urgent need for something definitive for the ecosystem. So part of the challenge was coming up with a definitive standard for DAOs (and not just off-chain metadata standards for DAOs) without over-specifying them.